plugins/woocommerce-blocks/assets/js/editor-components/switch-to-classic-shortcode-button/index.tsx

```diff
@@ -35,15 +37,25 @@ export function SwitchToClassicShortcodeButton( {
 	const closeModal = () => setOpen( false );
 	const { undo } = useDispatch( coreStore );
 
-	const switchButtonLabel =
-		block === 'woocommerce/cart'
-			? __( 'Switch to classic cart', 'woocommerce' )
-			: __( 'Switch to classic checkout', 'woocommerce' );
+	const isCart = block === 'woocommerce/cart';
 
-	const snackbarLabel =
-		block === 'woocommerce/cart'
-			? __( 'Switched to classic cart.', 'woocommerce' )
-			: __( 'Switched to classic checkout.', 'woocommerce' );
+	const switchButtonLabel = isCart
+		? __( 'Switch to classic cart', 'woocommerce' )
+		: __( 'Switch to classic checkout', 'woocommerce' );
+
+	const snackbarLabel = isCart
+		? __( 'Switched to classic cart.', 'woocommerce' )
+		: __( 'Switched to classic checkout.', 'woocommerce' );
+
+	const notice =
+		type === 'incompatible' ? 'incompatible_notice' : 'generic_notice';
```

Comment:
Same here, invalid type data won't produce an error and might display the generic notice erroneously. This way the bug could be in the wild for a long time before it gets reported.

Reply:
`type` is a props of the `<SwitchToClassicShortcodeButton>` component. Currently, we're only using this component here: https://github.com/woocommerce/woocommerce/blob/2749cebf7f06b5dbfd45550a8dc4f65038ff9bb8/plugins/woocommerce-blocks/assets/js/editor-components/incompatible-extension-notice/index.tsx#L144-L147

Assuming that we'd merge this PR, we'd end up with these two usages in our codebase:

```ts
<SwitchToClassicShortcodeButton
  block={ block }
  clientId={ clientId }
  type={ 'incompatible' }
/>
```

```ts
<SwitchToClassicShortcodeButton
  block={ blockName }
  clientId={ clientId }
  type={ 'generic' }
/>
```

That said, we could safe guard the `type` prop by changing

```ts
interface SwitchToClassicShortcodeButtonProps {
	block: 'woocommerce/cart' | 'woocommerce/checkout';
	clientId: string;
	type: string;
}
```

to

```ts
interface SwitchToClassicShortcodeButtonProps {
	block: 'woocommerce/cart' | 'woocommerce/checkout';
	clientId: string;
	type: 'incompatible' | 'generic';
}
```

That way, we'll imediately see an TS error is providing an invalid type. I'll adjust this accordingly.